static/app/components/useButtonHook.tsx

```diff
@@ -0,0 +1,60 @@
+import {useCallback} from 'react';
+
+import HookStore from 'sentry/stores/hookStore';
+
+type Props = {
+  busy?: boolean;
+  disabled?: boolean;
+  onClick?: (e: React.MouseEvent) => void;
+};
+
+const useGetHook = () => {
+  const useButtonTracking = HookStore.get('react-hook:track-buttons')[0];
+  const useButtonClick = ({disabled, busy, onClick}: Props) =>
+    useCallback(
+      (e: React.MouseEvent) => {
+        // Don't allow clicks when disabled or busy
+        if (disabled || busy) {
+          e.preventDefault();
+          e.stopPropagation();
+          return;
+        }
+
+        if (typeof onClick !== 'function') {
+          return;
+        }
+
+        onClick(e);
+      },
+      [disabled, busy, onClick]
+    );
+
+  const handleClick = useButtonTracking || useButtonClick;
+
+  return handleClick;
+};
+
+export default function useButtonHook({
```

Comment:
We don't want this in the `components/` directory, we keep hooks under `utils/`. Although, since this is only used by the `button.tsx` (I assume it's not meant to be used by other components?), it might be best to keep it in that file so people don't assume it's reusable.

And a bit of a nit but I think we could give this a slightly better name like `useButtonClickHandler` since the hook only gets applied sometimes, but it always returns a click handler.

Reply:
Sounds good! i think evan brought up not wanting to add too much to the button.tsx file so maybe it is best to keep it in utils but mention that it shouldn't be used else where? will change the name!